_includes/markdown/PHP.md

```diff
@@ -726,6 +726,65 @@ if ( ! empty( $_POST['_wpnonce'] ) && wp_verify_nonce( $_POST['_wpnonce'], 'my_a
+
+Samuel Wood (Otto) put together a guide to WordPress internationalization best practices, and engineers should take time to familiarize themselves with its guidance: [Internationalization: You’re probably doing it wrong](http://ottopress.com/2012/internationalization-youre-probably-doing-it-wrong/)
+
+It's important to note that the strings passed to translation functions should always be literal strings, never variables or named constants, and code shouldn't use [string interpolation](https://en.wikipedia.org/wiki/String_interpolation#PHP) to inject values into either string. Most tools used to create translations rely on [GNU gettext](https://www.gnu.org/software/gettext/) scanning source code for translation functions. PHP code won't be interpreted, only scanned like it was a block of plain text and stored similarly. If WordPress's translation APIs can't find an exact match for a string inside the translation files, it won't be able to translate the string. Instead, use [printf formatting codes](https://en.wikipedia.org/wiki/Printf_format_string) inside the string to be translated and pass the translated version of that string to sprintf() to fill in the values.
+
+For example:
+```php
+// This will confuse translation software
+$string = __( "$number minutes left", 'plugin-domain' );
+// So will this
+define( 'MINUTES_LEFT', '%d minutes left' );
+$string = __( MINUTES_LEFT, 'plugin-domain' );
+// Correct way to do a simple translation 
+$string = sprintf( __( '%d minutes left', 'plugin-domain' ), $number );
+// A more complex translation using _n() for plurals
+$string = sprintf( _n( '%d minute left', '%d minutes left',  $number, 'plugin-domain' ), $number );
+```
+
+Localizing a project differs from the core approach in two distinct ways:
+* A unique text domain should be used with all localization functions
+* Internationalized output should always be escaped
+##### Text Domains
+
+Each project should leverage a unique text domain for its strings. Text domains should be lowercase, alphanumeric, and use hyphens to separate multiple words: `tenup-project-name`.
+
+Like the translated strings they accompany, text domains should never be stored in a variable or named constant when used with core localization functions, as this practice can often produce unexpected results. Most tools used to create translation rely on [GNU gettext](https://www.gnu.org/software/gettext/) scanning source code for translation functions. PHP code won't be interpreted, only scanned like it was a block of plain text, and translation tools won't be able to assign the text domain correctly if it's not there in plain text.
```

Comment:
" Most tools used to create translation rely on [GNU gettext](https://www.gnu.org/software/gettext/) scanning source code for translation functions." that exact sentence appears twice.